contracts/BaseStrategy.sol

```diff
@@ -608,26 +609,23 @@ abstract contract BaseStrategy {
         emit Harvested(profit, loss, debtPayment, debtOutstanding);
     }
 
-    /**
-     * Liquidate as many assets as possible to `want`, irregardless of
-     * slippage, up to `_amountNeeded`. Any excess should be re-invested
-     * here as well.
-     */
-    function liquidatePosition(uint256 _amountNeeded) internal virtual returns (uint256 _amountFreed);
-
     /**
      * @notice
      *  Withdraws `_amountNeeded` to `vault`.
      *
      *  This may only be called by the Vault.
      * @param _amountNeeded How much `want` to withdraw.
+     * @return loss Any realized losses
      */
-    function withdraw(uint256 _amountNeeded) external {
+    function withdraw(uint256 _amountNeeded) external returns (uint256 loss) {
         require(msg.sender == address(vault), "!vault");
         // Liquidate as much as possible to `want`, up to `_amount`
-        uint256 amountFreed = liquidatePosition(_amountNeeded);
+        uint256 amountFreed;
+        (amountFreed, loss) = liquidatePosition(_amountNeeded);
         // Send it directly back (NOTE: Using `msg.sender` saves some gas here)
         want.transfer(msg.sender, amountFreed);
+        // Reinvest anything leftover
+        adjustPosition(vault.debtOutstanding());
```

Comment:
should this be a concern of tend instead of withdraw to reinvest leftover?

Reply:
good point, this could also be abused as an attack vector to force a strategy to rebalance itself, potentially into a bad position.